contracts/libraries/IterableOrderedOrderSet.sol

```diff
@@ -88,7 +97,10 @@ library IterableOrderedOrderSet {
         return true;
     }
 
-    function remove(Data storage self, bytes32 elementToRemove)
+    /// The element is removed from the linked list, but the node retains
+    /// information on which predecessor it had, so that a node in the chain
+    /// can be reached by following the predecessor chain of deleted elements.
+    function removeKeepHistory(Data storage self, bytes32 elementToRemove)
```

Comment:
Just thinking out loud:

Instead of having two remove functions, we could also have just one:

```
function remove(Data storage self, bytes32 elementToRemove, bool keepHistory)
```

I don't know which solution I would prefer.

Reply:
The code is very similar but they have very different roles, so I personally prefer to have two separate names to avoid mistakes.

Reply:
Cool. Then lets stick to your version!